src/ralph/backends/data/s3.py

```diff
@@ -192,63 +181,73 @@ def read(  # noqa: PLR0913
 
     def _read_bytes(
         self,
-        query: BaseQuery,
+        query: str,
         target: Optional[str],
         chunk_size: int,
         ignore_errors: bool,  # noqa: ARG002
     ) -> Iterator[bytes]:
         """Method called by `self.read` yielding bytes. See `self.read`."""
         target = self.default_bucket_name if target is None else target
-        response = self._get_object(target, query.query_string)
+        response = self._get_object(target, query)
         try:
             yield from response["Body"].iter_chunks(chunk_size)
         except (ReadTimeoutError, ResponseStreamingError) as err:
             msg = "Failed to read chunk from object %s"
-            self.logger.error(msg, query.query_string)
-            raise BackendException(msg % (query.query_string)) from err
+            self.logger.error(msg, query)
+            raise BackendException(msg % (query)) from err
 
         # Archive fetched, add a new entry to the history.
         self.append_to_history(
             {
                 "backend": self.name,
                 "action": "read",
-                "id": target + "/" + query.query_string,
+                "id": target + "/" + query,
                 "size": response["ContentLength"],
                 "timestamp": now(),
             }
         )
 
     def _read_dicts(
         self,
-        query: BaseQuery,
+        query: str,
         target: Optional[str],
         chunk_size: int,
         ignore_errors: bool,
     ) -> Iterator[dict]:
         """Method called by `self.read` yielding dictionaries. See `self.read`."""
         target = self.default_bucket_name if target is None else target
-        response = self._get_object(target, query.query_string)
+        response = self._get_object(target, query)
         try:
             lines = response["Body"].iter_lines(chunk_size)
             yield from parse_iterable_to_dict(lines, ignore_errors, self.logger)
         except (ReadTimeoutError, ResponseStreamingError) as err:
             msg = "Failed to read chunk from object %s"
-            self.logger.error(msg, query.query_string)
-            raise BackendException(msg % (query.query_string)) from err
+            self.logger.error(msg, query)
+            raise BackendException(msg % (query)) from err
 
         # Archive fetched, add a new entry to the history.
         self.append_to_history(
             {
                 "backend": self.name,
                 "action": "read",
-                "id": target + "/" + query.query_string,
+                "id": target + "/" + query,
                 "size": response["ContentLength"],
                 "timestamp": now(),
             }
         )
 
-    def _get_object(self, bucket: str, key: str) -> dict:
-        """Return S3 object wrapping the exception."""
+    def _get_object(self, bucket: Optional[str], key: str) -> dict:
+        """Validate bucket and key and return S3 object wrapping the exception."""
```

Comment:
```suggestion
        """Validate bucket (target) and key (query) and return the S3 object."""
```
I don't understand "wrapping the exception"